components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/OAuthUserConsentedScopesDAOImpl.java

```diff
@@ -0,0 +1,411 @@
+                    ps.setBoolean(4, false);
+                    ps.addBatch();
+                }
+            }
+            ps.executeBatch();
+        }
+    }
+
+    private void deleteUserConsent(Connection connection, String userId, String appId, int tenantId)
+            throws SQLException {
+
+        try (PreparedStatement ps = connection.prepareStatement(SQLQueries.REMOVE_OAUTH2_USER_CONSENT_FOR_APP)) {
+            ps.setString(1, userId);
+            ps.setString(2, appId);
+            ps.setInt(3, tenantId);
+            ps.execute();
+        }
+    }
+
+    private String generateConsentId() {
+
+        return UUID.randomUUID().toString();
+    }
+
+    private String getConsentId(Connection connection, String userId, String appId, int tenantId) throws SQLException {
+
+        String consentId = null;
+        try (PreparedStatement ps = connection.prepareStatement(SQLQueries.GET_CONSENT_ID_FOR_CONSENT)) {
+            ps.setString(1, userId);
+            ps.setString(2, appId);
+            ps.setInt(3, tenantId);
+            try (ResultSet rs = ps.executeQuery()) {
+                while (rs.next()) {
+                    consentId = rs.getString(1);
+                }
+            }
+        }
+        return consentId;
+    }
+
+    private void validateUserConsent(UserApplicationScopeConsentDO userConsent)
+            throws IdentityOAuth2ScopeConsentClientException {
+
+        if (userConsent == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("User consent can't be null or empty.");
+            }
+            throw new IdentityOAuth2ScopeConsentClientException("User consent can't be null or empty.");
+        }
+
+        if (StringUtils.isBlank(userConsent.getAppId())) {
+            if (log.isDebugEnabled()) {
+                log.debug("Application Id can't be null/empty.");
+            }
+            throw new IdentityOAuth2ScopeConsentClientException("Application Id can't be null/empty.");
+        }
+
+        if (CollectionUtils.isEmpty(userConsent.getApprovedScopes())
+                && CollectionUtils.isEmpty(userConsent.getDisapprovedScopes())) {
+            if (log.isDebugEnabled()) {
+                log.debug("User hasn't approved or disapproved any scopes.");
+            }
+            throw new IdentityOAuth2ScopeConsentClientException("User hasn't approved or disapproved any scopes.");
+        }
+    }
+
+    private void validateAppId(String appId) throws IdentityOAuth2ScopeConsentClientException {
+
+        if (StringUtils.isBlank(appId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Application Id can't be null/empty.");
+            }
+            throw new IdentityOAuth2ScopeConsentClientException("Application Id can't be null/empty.");
+        }
+    }
+
+    private void validateUserId(String userId) throws IdentityOAuth2ScopeConsentClientException {
+
+        if (StringUtils.isBlank(userId)) {
```

Comment:
Don't we need to validate the userId from the userstore level? It does not happen in the osgi level as well. Better to do that in the osgi level

Reply:
Since we're resolving the userID from the userstore. We don't need to check this in here.